addons/project_issue_sheet/project_issue_sheet.py

```diff
@@ -21,13 +21,25 @@
 
 from openerp.osv import fields,osv,orm
 from openerp.tools.translate import _
+from openerp.addons.analytic.models import analytic
 
 class project_issue(osv.osv):
     _inherit = 'project.issue'
     _description = 'project issue'
+    
+    def _get_current_contract_state(self, cr, uid, ids, field_names, args, context=None):
+        res = {}
+        for issue in self.browse(cr, uid, ids, context=context):
+            res[issue.id] = False
+            contrat = issue.project_id and issue.project_id.analytic_account_id
+            if contrat.partner_id and contrat.use_issues:
+                res[issue.id] = contrat.state
+        return res
+        
     _columns = {
         'timesheet_ids': fields.one2many('hr.analytic.timesheet', 'issue_id', 'Timesheets'),
         'analytic_account_id': fields.many2one('account.analytic.account', 'Analytic Account'), 
+        'contract_state': fields.function(_get_current_contract_state, string='Contract Status', type='selection', selection=analytic.ANALYTIC_ACCOUNT_STATE), 
```

Comment:
Why would you have a `project.issue` linked to a project with `use_issues` at `False`, does not make much sense. Also you why would you get no state if there is no partner. All these are display conditions that should be in the view, not on the model. So a related should be enough.